docs/en/ingest-management/fleet/add-fleet-server-kubernetes-content.asciidoc

```diff
@@ -0,0 +1,176 @@
+// tag::quickstart-secret[]
+The following command assumes you have the {es} CA file available in a local file.
++
+[source, shell]
+------------------------------------------------------------
+kubectl create secret generic fleet-server-certs \
+  --from-file=es-ca.crt=<PATH_TO_ES_CA_CERT_FILE> <1>
+------------------------------------------------------------
+<1> Substitute `<PATH_TO_ES_CA_CERT_FILE>` with your local file containing the {es} CA(s).
++
+If you prefer to obtain a `yaml manifest` of the Secret to create, append `--dry-run=client -o=yaml` to the command and save the output to a file.
+// end::quickstart-secret[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::production-secret[]
+The following command assumes you have the {es} CA file, the {fleet-server} certificate, and the {fleet-server} key available in local files.
++
+[source, shell]
+------------------------------------------------------------
+kubectl create secret generic fleet-server-certs \
+  --from-file=es-ca.crt=<PATH_TO_ES_CA_CERT_FILE> \ <1>
+  --from-file=fleet-server.crt=<PATH_TO_FLEET_SERVER_CERT> \ <2>
+  --from-file=fleet-server.key=<PATH_TO_FLEET_SERVER_CERT_KEY> <3>
+------------------------------------------------------------
+<1> Substitute `<PATH_TO_ES_CA_CERT_FILE>` with your local file containing the {es} CA(s).
+<2> Substitute `<PATH_TO_FLEET_SERVER_CERT>` with your local file containing the server TLS certificate for the {fleet-server}.
+<3> Substitute `PATH_TO_FLEET_SERVER_CERT_KEY` with your local file containing the server TLS key for the {fleet-server}.
++
+If you prefer to obtain a `yaml manifest` of the Secret to create, append `--dry-run=client -o=yaml` to the command and save the output to a file.
+// end::production-secret[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::quickstart-deployment[]
+[source, yaml]
+------------------------------------------------------------
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: fleet-server
+spec:
+  replicas: 1 <1>
+  selector:
+    matchLabels:
+      app: fleet-server
+  template:
+    metadata:
+      labels:
+        app: fleet-server
+    spec:
+      automountServiceAccountToken: false <2>
+      containers:
+      - name: elastic-agent
+        image: docker.elastic.co/beats/elastic-agent:{version}
```

Comment:
@bmorelli25 , @kilfoyle : Is there any way (`attribute` or hack) to make the `version` to be automatically rendered to the right version? For example in `current`, today it should be rendered as `8.16.1`.

Reply:
@eedugon  You found it! That `{version}` attribute will resolve to the current version in the docs. We use it [here](https://github.com/elastic/ingest-docs/blob/1253e201a23639ff38ff31ee5ae93f88733fac69/docs/en/ingest-management/beats-agent-comparison.asciidoc?plain=1#L68), for example.

Reply:
oh! thanks @kilfoyle!
Is it normal then that the {version} is not rendered on previews and in local builds?

because the current preview is still showing it as {version}, and I was a bit scared because of it :)

Reply:
Good question! You're so close. You have to explicitly tell the codeblock to use attribute substitution. To do this, you use `[... subs="attributes"]`. I'll add a comment showing this.

Reply:
https://github.com/elastic/ingest-docs/pull/1518/files#r1878558170

Reply:
all clear now, thanks!